src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/UniversalRemoteAuthenticationProvider.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages
+{
+    /// <summary>
+    /// Implementation of IRemoteAuthenticationProvider for the "Universal" authentication type.
+    /// </summary>
+    [Export(typeof(IRemoteAuthenticationProvider))]
+    [AppliesTo("SupportUniversalAuthentication")]
+    internal class UniversalRemoteAuthenticationProvider : IRemoteAuthenticationProvider
+    {
+        private static readonly Guid s_universalPortSupplier = new Guid("EE56E4E8-E866-4915-A18E-1DE7114BD7BB");
+
+        [ImportingConstructor]
+        public UniversalRemoteAuthenticationProvider()
+        {
+        }
+
+        public string DisplayName => "Universal";
+
+        public string Name => "None";
+
+        public Guid PortSupplierGuid => VSConstants.DebugPortSupplierGuids.NoAuth_guid;
```

Comment:
Explain the different between AuthMode and PortSupplier?

Reply:
I don't understand how "universal" gets turned on - it just passed NoAuth as the PortSupplier to the debugger the same way that NoAuthentication does.

Reply:
AuthMode and PortSupplier are different, and I think its a bug that the parameter name for the Remote Discovery Dialog is called "portSupplier". The term is used for the combination of how msvsmon will be deployed, and how to connect to it. So when using Universal authentication mode, thats the trigger to deploy msvsmon and start it, but no authentication is required for the connection to it.  Obviously the deployment work is not part of this PR which is why this PR doesn't turn on Universal as an option.